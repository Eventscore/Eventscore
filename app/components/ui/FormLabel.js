//<FormLabel></FormLabel>
import React, { Component, PropTypes } from 'react';
import { FormLabel } from 'react-native-elements';

// Consts and Libs
import { AppColors, AppFonts } from '@theme/';

class CustomFormLabel extends Component {
  static propTypes = {
    labelStyle: PropTypes.oneOfType([
      PropTypes.array,
      PropTypes.shape({}),
    ]),
    children: PropTypes.node,
  }

  static defaultProps = {
    containerStyle: [],
    labelStyle: [],
    children: null,
  }

  labelProps = () => {
    // Defaults
    const props = {
      ...this.props,
      labelStyle: [{
        color: AppColors.textPrimary,
        fontFamily: AppFonts.base.family,
      }],
    };

    if (this.props.labelStyle) {
      props.labelStyle.push(this.props.labelStyle);
    }

    return props;
  }

  render = () => <FormLabel {...this.labelProps()}>{this.props.children}</FormLabel>;
}

export default CustomFormLabel;
