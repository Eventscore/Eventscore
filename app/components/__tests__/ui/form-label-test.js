/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import FormLabel from '@ui/FormLabel';

it('FormLabel renders correctly', () => {
  const tree = renderer.create(
    <FormLabel>John Smith</FormLabel>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
