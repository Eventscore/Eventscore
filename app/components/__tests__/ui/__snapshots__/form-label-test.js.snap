// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabel renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": undefined,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Array [
          Object {
            "color": "#86939e",
            "fontSize": 15,
            "fontWeight": "bold",
            "marginBottom": 1,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 15,
          },
          Array [
            Object {
              "color": "#222222",
              "fontFamily": "HelveticaNeue",
            },
            Array [],
          ],
          undefined,
        ],
      ]
    }
  >
    John Smith
  </Text>
</View>
`;
